wrapper/src/main/java/software/amazon/jdbc/dialect/AuroraMysqlDialect.java

```diff
@@ -105,5 +112,23 @@ public HostListProviderSupplier getHostListProvider() {
           IS_READER_QUERY);
     };
   }
+
+  @Override
+  public String getBlueGreenStatusQuery() {
+    return BG_STATUS_QUERY;
+  }
+
+  @Override
+  public boolean isStatusAvailable(final Connection connection) {
+    try {
+      try (Statement statement = connection.createStatement();
+          ResultSet rs = statement.executeQuery(TOPOLOGY_TABLE_EXIST_QUERY)) {
+        return rs.next();
+      }
+    } catch (SQLException ex) {
+      return false;
+    }
```

Comment:
```suggestion
    try (Statement statement = connection.createStatement();
        ResultSet rs = statement.executeQuery(TOPOLOGY_TABLE_EXIST_QUERY)) {
      return rs.next();
    } catch (SQLException ex) {
      return false;
    }
```

Reply:
This try-with-resources doesn't catch exceptions during initializing of these resources. We need an extra try-catch.